pygsp/operators/reduction.py

```diff
@@ -327,27 +342,24 @@ def kron_reduction(G, ind):
     N = np.shape(L)[0]
     ind_comp = np.setdiff1d(np.arange(N, dtype=int), ind)
 
-    L_red = L[np.ix_(ind, ind)]
-    L_in_out = L[np.ix_(ind, ind_comp)]
-    L_out_in = L[np.ix_(ind_comp, ind)].tocsc()
-    L_comp = L[np.ix_(ind_comp, ind_comp)].tocsc()
+    L_red = extract_submatrix(L,ind, ind)
+    L_in_out = extract_submatrix(L, ind, ind_comp)
+    L_out_in = L_in_out.transpose().tocsc()
+    L_comp = extract_submatrix(L,ind_comp, ind_comp).tocsc()
 
-    Lnew = L_red - L_in_out.dot(spsolve(L_comp, L_out_in))
+    Lnew = L_red - L_in_out.dot(splu_inv_dot(L_comp, L_out_in))
 
-    # Make the laplacian symmetric if it is almost symmetric!
-    if np.abs(Lnew - Lnew.T).sum() < np.spacing(1) * np.abs(Lnew).sum():
-        Lnew = (Lnew + Lnew.T) / 2.
+    # Threshold excedingly small values for stability
+    Lnew = Lnew.tocoo()
+    Lnew.data[abs(Lnew.data) < threshold] = 0
+    Lnew = Lnew.tocsc()
+    Lnew.eliminate_zeros()
+
+    # Enforces symmetric Laplacian
+    Lnew = (Lnew + Lnew.T) / 2.
```

Comment:
Do we always want the Laplacian to be symmetric here? In the previous implementation, this line was called only under the conditional statement.